template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/hbase/HbaseCloudStorageServiceConfigProvider.java

```diff
@@ -56,11 +57,12 @@ public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, Te
         String cdhVersion = getCdhVersion(source);
         boolean is727OrNewer = isVersionNewerOrEqualThanLimited(cdhVersion, CLOUDERAMANAGER_VERSION_7_2_7);
         String accountId = ThreadBasedUserCrnProvider.getAccountId();
-        boolean sdxHbaseCloudStorageEnabled =
-                entitlementService.sdxHbaseCloudStorageEnabled(accountId);
+        boolean sdxHbaseCloudStorageEnabled = entitlementService.sdxHbaseCloudStorageEnabled(accountId);
+        boolean razEnabled =  source.getGeneralClusterConfigs().isEnableRangerRaz();
         return source.getFileSystemConfigurationView().isPresent()
                 && cmTemplateProcessor.isRoleTypePresentInService(getServiceType(), getRoleTypes())
-                && (!datalakeCluster || (is727OrNewer && sdxHbaseCloudStorageEnabled));
+                && (!datalakeCluster || (is727OrNewer && ((CloudPlatform.AWS.equals(source.getCloudPlatform()) && !razEnabled) ||
```

Comment:
There is still an unnecessary pair of parentheses in `(CloudPlatform.AWS.equals(source.getCloudPlatform())`.

Reply:
Updated to make the logic cleaner